test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework.services.docker;
+
+
+import com.google.common.base.Preconditions;
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.NetworkConfig;
+import com.spotify.docker.client.messages.swarm.Service;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import io.pravega.common.concurrent.FutureHelpers;
+import lombok.extern.slf4j.Slf4j;
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Slf4j
+public abstract class DockerBasedService  implements io.pravega.test.system.framework.services.Service {
+
+    static final int ZKSERVICE_ZKPORT = 2181;
+    static final int CONTROLLER_PORT = 9092;
+    static final String IMAGE_PATH = System.getProperty("dockerImageRegistry");
+    static final String PRAVEGA_VERSION = System.getProperty("imageVersion");
+    DockerClient dockerClient;
+    String serviceName;
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(3);
+
+    DockerBasedService(String serviceName) {
+        dockerClient = DefaultDockerClient.builder().uri("http://"+ System.getProperty("masterIP")+ ":2375").build();
+        try {
+            if (dockerClient.listNetworks(DockerClient.ListNetworksParam.byNetworkName("docker-network")).isEmpty()) {
+                dockerClient.createNetwork(NetworkConfig.builder().driver("overlay").name("docker-network").build());
+            }
+        } catch (DockerException | InterruptedException e) {
+            log.error("Unable to create network {}", e);
+        }
+        this.serviceName = serviceName;
+        }
+
+
+    @Override
+    public String getID() {
+        Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+        String serviceId = null;
+        try {
+            List<Service> serviceList = dockerClient.listServices(criteria);
+            serviceId = serviceList.get(0).id();
+        }  catch (DockerException | InterruptedException  e) {
```

Comment:
This is not the proper way to handle interruptedException. We have a helper method in common.

Reply:
Changed to use helper method.